src/unittest/test.h

```diff
@@ -86,8 +114,26 @@ class TestFailedException : public std::exception {
 	}                                                                     \
 }
 
+// Asserts the comparison specified by CMP is true, or fails the current unit test
+#define UASSERTCMP_EXP(T, CMP, actual, expected) {                            \
+	T a = (actual);                                                       \
+	T e = (expected);                                                     \
+	if (!(a CMP e)) {                                                     \
+		rawstream                                                         \
+			<< "Test assertion failed (but expected): " << #actual << " " << #CMP << " " \
+			<< #expected << std::endl                                     \
+			<< "    at " << fs::GetFilenameFromPath(__FILE__) << ":"      \
+			<< __LINE__ << std::endl                                      \
+			<< "    actual:   " << a << std::endl << "    expected: "     \
+			<< e << std::endl;                                            \
+		throw TestExpFailedException();                                      \
+	}                                                                     \
```

Comment:
I think it'd be better to have a wrapper that converts TestFailedException to TestExpFailedException than duplicating all this.

Reply:
Great suggestion, sfan5. I moved the expected failure handling into a wrapper macro.